sdk/eventgrid/azure-messaging-eventgrid-namespaces/src/main/java/com/azure/messaging/eventgrid/namespaces/EventGridReceiverAsyncClient.java

```diff
@@ -385,12 +387,48 @@ public Mono<ReceiveResult> receive(Integer maxEvents, Duration maxWaitTime) {
             requestOptions.addQueryParam("maxEvents", String.valueOf(maxEvents), false);
         }
         if (maxWaitTime != null) {
+            // Ensure the http client timeout is longer than the maxWaitTime
+            addTimeoutToContext(requestOptions, maxWaitTime);
             requestOptions.addQueryParam("maxWaitTime", String.valueOf(maxWaitTime.getSeconds()), false);
         }
         return receiveWithResponse(topicName, subscriptionName, requestOptions).flatMap(FluxUtil::toMono)
             .map(protocolMethodData -> protocolMethodData.toObject(ReceiveResult.class));
     }
 
+    /**
+     * Receive a batch of Cloud Events from a subscription.
+     *
+     * @param maxEvents Max Events count to be received. Minimum value is 1, while maximum value is 100 events. If not
+     * specified, the default value is 1.
+     * @param maxWaitTime Max wait time value for receive operation in Seconds. It is the time in seconds that the
```

Comment:
Since we are using `Duration` we should not specify the units. What is the behavior if user provides `Duration.ofMillis(500)`? Do we let the service throw an error for this case?

Reply:
This comment comes from the [tsp](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/eventgrid/Azure.Messaging.EventGrid/main.tsp#L287). Further the field is an `int32` so we couldn't pass a fractional second. I think the comment is reasonable. I could extend it to explain we'll ignore fractional values if you think that's necessary, but I'm inclined to keep the tsp comment.

We call `getSeconds()` on the `Duration` we're given, which rounds down, so we always end up with a valid seconds value.

Service returns 400 if the maxWaitTime is outside the 10-120s bound.